frontend/integration-tests/views/olm-catalog.view.ts

```diff
@@ -1,12 +1,12 @@
 /* eslint-disable no-undef, no-unused-vars */
 
-import { browser, $, $$, by, ExpectedConditions as until } from 'protractor';
+import { browser, $, $$, by, element, ExpectedConditions as until } from 'protractor';
 
 import * as crudView from './crud.view';
 
-export const entryRows = $$('.co-resource-list__item');
-export const entryRowFor = (name: string) => entryRows.filter((row) => row.$('.co-clusterserviceversion-logo__name__clusterserviceversion').getText()
-  .then(text => text === name || text.startsWith(name))).first();
+const rowSelector = '.co-resource-list__item';
+export const entryRows = $$(rowSelector);
+export const entryRowFor = (name: string) => element(by.cssContainingText(rowSelector, name));
```

Comment:
This is less precise, but it seems unlikely we'd have false matches. It should avoid the `StaleElementReferenceError`. I verified it correctly fails if the item is missing:

```
Failed Specs:

1) Installing a service from a Catalog Source : displays Catalog Source with expected available packages
   NoSuchElementError: No element found using locator: by.cssContainingText(".co-resource-list__item", "BOGUS")
```